Gems/PhysX/Code/Tests/PhysXMeshColliderComponentModeTests.cpp

```diff
@@ -0,0 +1,115 @@
+        auto colliderEntity = CreateMeshColliderComponent();
+        AZ::Vector3 assetScale(10.0f, 10.0f, 10.0f);
+        colliderEntity->FindComponent<TestMeshColliderComponent>()->SetAssetScale(assetScale);
+
+        AzToolsFramework::SelectEntity(colliderEntity->GetId());
+        EnterComponentMode<TestMeshColliderComponent>();
+
+        PhysX::ColliderComponentModeRequestBus::Broadcast(&PhysX::ColliderComponentModeRequests::SetCurrentMode,
+            PhysX::ColliderComponentModeRequests::SubMode::Dimensions);
+
+        // When the 'R' key is pressed
+        QTest::keyPress(&m_editorActions.m_componentModeWidget, Qt::Key_R);
+
+        // Then the asset scale should be reset.
+        assetScale = colliderEntity->FindComponent<TestMeshColliderComponent>()->GetAssetScale();
+        EXPECT_THAT(assetScale, IsClose(AZ::Vector3::CreateOne()));
+    }
+
+    using PhysXMeshColliderComponentModeManipulatorTest =
+        UnitTest::IndirectCallManipulatorViewportInteractionFixtureMixin<PhysXMeshColliderComponentModeTest>;
+
+    TEST_F(PhysXMeshColliderComponentModeManipulatorTest, AssetScaleManipulatorsScaleInCorrectDirection)
+    {
+        auto colliderEntity = CreateMeshColliderComponent();
+        colliderEntity->FindComponent<TestMeshColliderComponent>()->SetAssetScale(AZ::Vector3::CreateOne());
+        EnterComponentMode<TestMeshColliderComponent>();
+        PhysX::ColliderComponentModeRequestBus::Broadcast(&PhysX::ColliderComponentModeRequests::SetCurrentMode,
+            PhysX::ColliderComponentModeRequests::SubMode::Dimensions);
+
+        // position the camera so the X axis manipulator will be flipped
+        AzFramework::SetCameraTransform(
+            m_cameraState,
+            AZ::Transform::CreateFromQuaternionAndTranslation(
+                AZ::Quaternion::CreateRotationZ(-AZ::Constants::QuarterPi), AZ::Vector3(-5.0f, -5.0f, 0.0f)));
+
+        // select a point in world space slightly displaced from the position of the entity in the negative x direction
+        // in order to grab the X manipulator
+        const float x = 0.1f;
+        const float xDelta = 0.1f;
+        const AZ::Vector3 worldStart(-x, 0.0f, 0.0f);
+
+        // position in world space to drag to
+        const AZ::Vector3 worldEnd(-(x + xDelta), 0.0f, 0.0f);
+
+        DragMouse(m_cameraState, m_actionDispatcher.get(), worldStart, worldEnd);
+
+        const auto worldToScreenMultiplier = 1.0f / AzToolsFramework::CalculateScreenToWorldMultiplier(worldStart, m_cameraState);
+        const auto assetScale = colliderEntity->FindComponent<TestMeshColliderComponent>()->GetAssetScale();
+        // need quite a large tolerance because using screen co-ordinates limits precision
+        const float tolerance = 0.01f;
+        EXPECT_NEAR(assetScale.GetX(), 1.0f + xDelta * worldToScreenMultiplier, tolerance);
+        EXPECT_NEAR(assetScale.GetY(), 1.0f, tolerance);
+        EXPECT_NEAR(assetScale.GetZ(), 1.0f, tolerance);
```

Comment:
nit: Consider using `EXPECT_THAT` and the `IsClose` matcher for this, it's a slightly shorter/cleaner way to express the same thing (and you still get useful output if the test fails)

Reply:
will change.